ddtrace/tracer/option.go

```diff
@@ -145,16 +146,23 @@ func WithHTTPRoundTripper(r http.RoundTripper) StartOption {
 // WithAnalytics allows specifying whether Trace Search & Analytics should be enabled
 // for integrations.
 func WithAnalytics(on bool) StartOption {
-	if on {
-		return WithAnalyticsRate(1.0)
+	return func(cfg *config) {
+		if on {
+			globalconfig.SetAnalyticsRate(1.0)
```

Comment:
... like you did here 😄

Reply:
In this case, I don't have the direct ability to assign the value, so a function call is actually needed.

Reply:
Right, sorry about that.